### PR TITLE
Feature - Sign and submit page

### LINKED ISSUE #3325: Sign and Submit page (Read Only)
# Description

[Figma Prototype](https://www.figma.com/proto/N9NoPunkqaynjsi89Wrbh0/Direct-Intake---Poster-and-application?node-id=229%3A5735&starting-point-node-id=226%3A4567)

This page should be rendered at the url `applications/:id/submit`.

Build the Step 2 page of the application.
Make use of the Application Wrapper #3323
It is not necessary to add functionality yet to the Submit button. ~~However, the submit button should be disabled if the Profile is incomplete, or if any required, occupational skills are missing.~~  This functionality will be added with draft applications.

Note: The Transferable/Occupational category comes from the Skill Family, which skills may have several of. If a skill has zero skill families, list it here. Only  Consider a skill to be Transferable (and therefore unlisted) if it has at least one SkillFamily and all are Transferable.

image

## Acceptance Criteria

- [ ] this page is rendered at the url `applications/:id/submit`
- [ ] details of the current pool appear correctly in the wrapper
- [ ] the submit button is disabled if the user's profile is incomplete or if missing any required occupational skills

### PR DESCRIPTION
**Pull request by yonikid15**
Resolves #3325 

## Notes
- Creates SignAndSubmit page. This page is read-only and can be viewed on Storybook.
- Also made some responsive style changes to ApplicationWrapper.

## Acceptance Criteria

- [x] this page is rendered at the url `applications/:id/submit`
- [x] details of the current pool appear correctly in the wrapper
- [x] the submit button is disabled if the user's profile is incomplete or if missing any required occupational skills

### CODE REVIEW

File: frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+    }),
+  ];
+  return (
+    <FormProvider {...methods}>
+      <form
+        onSubmit={methods.handleSubmit(() => {
+          // TODO: ADD SUBMIT APPLICATION MUTATION HERE
+        })}
+      >
+        <p>You made it! By signing your name below you confirm that:</p>
```
Comment (esizer) [inline frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx:83]: This should be wrapped in `intl.formatMessage()`

File: frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+              {item.component}
+            </TableOfContents.Section>
+          ))}
+        </TableOfContents.Content>
+      </TableOfContents.Wrapper>
+    </ApplicationPageWrapper>
+  );
+};
+
+const SignAndSubmitPage: React.FC<{ id: string }> = ({ id }) => {
```
Comment (esizer) [inline frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx:251]: `React.FC` is not longer reccomened and we are slowly moving to:

`const Component = ({props}: PropsInterface) => {}`

File: frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+      {steps.map((item) => (
+        <li key={uniqueId()} data-h2-margin="b(bottom, m)">
+          {item}
+        </li>
+      ))}
+    </ol>
+  );
+};
+
+const Signature: React.FC<{ isNotComplete: boolean }> = ({ isNotComplete }) => {
```
Comment (esizer) [inline frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx:56]: See: The other comment about `React.FC`

File: frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+import Pending from "@common/components/Pending";
+import TableOfContents from "@common/components/TableOfContents";
+import { commonMessages, errorMessages } from "@common/messages";
+import { ArrowSmRightIcon } from "@heroicons/react/outline";
+import {
+  BriefcaseIcon,
+  ClipboardCheckIcon,
+  UserIcon,
+} from "@heroicons/react/solid";
+import { uniqueId } from "lodash";
```
Comment (esizer) [inline frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx:14]: I think the preferred way of importing lodash is:

```tsx
import uniqueId from "lodash/uniqueId";
```

**Review by esizer — CHANGES_REQUESTED** [review]
Everything looks great! I just left some comments related to some updated patterns we introduced recently and a `intl` thing.

File: frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx (hunk 1 above)
Comment (yonikid15) [reply frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx]: Fixed by [Wrap string in formattedMessage](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3678/commits/06984e8ba58a46b0f58aed2d7a4cd923ea83bc04)

File: frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx (hunk 2 above)
Comment (yonikid15) [reply frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx:258]: Fixed by [Remove React.FC type from components](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3678/commits/c8632abf5980eac17565ab2f878ec1498f93d15e)

File: frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx (hunk 3 above)
Comment (yonikid15) [reply frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx]: Fixed by [Remove React.FC type from components](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3678/commits/c8632abf5980eac17565ab2f878ec1498f93d15e)

File: frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx (hunk 4 above)
Comment (yonikid15) [reply frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx]: Fixed by [Fix uniqueId lodash import](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3678/commits/54d52434c33723b6a09f1d74b30227a262f0977f)

File: frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import React from "react";
+import { Meta, Story } from "@storybook/react";
```
[the remark this answers is not in the document]
Comment (yonikid15) [reply frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.stories.tsx]: Fixed by [Remove deprecated storybook pattern](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3678/commits/bd6802d26c23e01c10757365330c8c0cd11f7ab2)

**Review by esizer — APPROVED** [review]
Nice, looks great! 👍

File: frontend/common/src/components/Breadcrumbs/Breadcrumbs.tsx [hunk 6]
```diff
@@ -11,41 +11,84 @@ export interface BreadcrumbsProps {
...
           <Link
             data-h2-display="base(flex)"
             data-h2-align-items="base(center)"
-            href={link.href}
-            key={link.title}
+            href={previousStep.href}
+            key={previousStep.title}
           >
-            {link.title}
+            {previousStep.icon || ""} {previousStep.title}
```
Comment (esizer) [inline frontend/common/src/components/Breadcrumbs/Breadcrumbs.tsx:53]: We recently removed icons from breadcrumbs so i think we can just use `{link.title}` now

File: frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx [hunk 7]
```diff
@@ -0,0 +1,291 @@
...
+      defaultMessage:
+        "You are still encouraged to keep your profile up to date, updated versions will be used at later steps of the hiring process.",
+      description: "Important info list item on sign and submit page.",
+    }),
+  ];
+
+  return (
+    <ol>
+      {steps.map((item) => (
+        <li key={uniqueId()} data-h2-margin="b(bottom, m)">
```
Comment (esizer) [inline frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx:48]: Will need to be updated to use new syntax:

`data-h2-margin="base(0, 0, x1, 0)"`

File: frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx [hunk 8]
```diff
@@ -0,0 +1,291 @@
...
+          })}
+        </p>
+        <ul>
+          {confirmations.map((item) => (
+            <li key={uniqueId()} data-h2-margin="b(bottom, m)">
+              {item}
+            </li>
+          ))}
+        </ul>
+        <div data-h2-width="b(100) s(75) m(50)" data-h2-margin="b(bottom, m)">
```
Comment (esizer) [inline frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx:98]: Will need to be updated to use new syntax.

File: frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx [hunk 9]
```diff
@@ -0,0 +1,291 @@
...
+          {intl.formatMessage({
+            defaultMessage:
+              "You made it! By signing your name below you confirm that:",
+            description:
+              "Confirmation message before signature form on sign and submit page.",
+          })}
+        </p>
+        <ul>
+          {confirmations.map((item) => (
+            <li key={uniqueId()} data-h2-margin="b(bottom, m)">
```
Comment (esizer) [inline frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx:93]: Will need to be updated to use new syntax.

File: frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx [hunk 10]
```diff
@@ -0,0 +1,291 @@
...
+            })}
+            type="text"
+            name="signature"
+            rules={{
+              required: intl.formatMessage(errorMessages.required),
+            }}
+            disabled={isNotComplete}
+          />
+        </div>
+        <div data-h2-text-align="b(center) s(left)">
```
Comment (esizer) [inline frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx:114]: Will need to be updated to use new syntax.

File: frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx [hunk 11]
```diff
@@ -0,0 +1,291 @@
...
+          <Submit
+            color="cta"
+            text={
+              <span data-h2-display="b(flex)" data-h2-align-items="b(center)">
```
Comment (esizer) [inline frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx:118]: Will need to be updated to use new syntax.

**Review by esizer — CHANGES_REQUESTED** [review]
Nice work! I just noticed a few places where the old hydrogen syntax still pops up as well as removing icons from breadcrumbs.

File: frontend/common/src/components/Breadcrumbs/Breadcrumbs.tsx (hunk 6 above)
Comment (yonikid15) [reply frontend/common/src/components/Breadcrumbs/Breadcrumbs.tsx]: Fixed by [Remove icon from breadcrumb link](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3678/commits/1fce5a392191f998122633106fb63454c14abde0)

File: frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx (hunk 7 above)
Comment (yonikid15) [reply frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx]: Fixed by [Update hydrogen utils and fix type error](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3678/commits/ef2a71837e961f068167149907476e257f0bc6f6)

File: frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx (hunk 8 above)
Comment (yonikid15) [reply frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx]: Fixed by [Update hydrogen utils and fix type error](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3678/commits/ef2a71837e961f068167149907476e257f0bc6f6)

File: frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx (hunk 9 above)
Comment (yonikid15) [reply frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx]: Fixed by [Update hydrogen utils and fix type error](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3678/commits/ef2a71837e961f068167149907476e257f0bc6f6)

File: frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx (hunk 10 above)
Comment (yonikid15) [reply frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx]: Fixed by [Update hydrogen utils and fix type error](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3678/commits/ef2a71837e961f068167149907476e257f0bc6f6)

File: frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx (hunk 11 above)
Comment (yonikid15) [reply frontend/talentsearch/src/js/components/signAndSubmit/SignAndSubmitPage.tsx]: Fixed by [Update hydrogen utils and fix type error](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3678/commits/ef2a71837e961f068167149907476e257f0bc6f6)

**Review by esizer — APPROVED** [review]
Looks good! 👍

**Pull request merged**